cpp/cmake/thirdparty/get_raft.cmake

```diff
@@ -36,19 +36,19 @@ function(find_and_configure_raft)
         string(APPEND RAFT_COMPONENTS " distributed")
     endif()
 
-
     if(PKG_COMPILE_LIBRARY)
-      string(APPEND RAFT_COMPONENTS " compiled")
+      if(NOT PKG_USE_RAFT_STATIC)
+        string(APPEND RAFT_COMPONENTS " compiled")
+        set(RAFT_COMPILED_LIB raft::compiled)
```

Comment:
function scope means that this variable won't exist outside of the caller. This needs to be `PARENT_SCOPE` so that `cpp/CMakeLists.txt` has a value.

```suggestion
        set(RAFT_COMPILED_LIB raft::compiled PARENT_SCOPE)
```